### ISSUE TITLE
System.ArgumentOutOfRangeException: 'Specified argument was out of the range of valid values. Parameter name: index'

### ISSUE THREAD

**Issue by Leon99** [reactiveui/refit#737]
```c#
        [Post("/transactions/{id}/confirm")]
        Task ConfirmTransaction(int id, [Query][AliasAs("send_to_payer")] bool sendToPayer, [Header("Authorization")] string authorization);
```
started to throw after updating to 4.7.51
```
   at System.Linq.Enumerable.ElementAt[TSource](IEnumerable`1 source, Int32 index)
   at Refit.RequestBuilderImplementation.<>c__DisplayClass17_0.<<BuildRequestFactoryForMethod>b__0>d.MoveNext() in d:\a\1\s\Refit\RequestBuilderImplementation.cs:line 594
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Refit.RequestBuilderImplementation.<>c__DisplayClass20_0.<<BuildVoidTaskFuncForMethod>b__0>d.MoveNext() in d:\a\1\s\Refit\RequestBuilderImplementation.cs:line 730
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at NexPay.ClientPortal.Services.Payments.PaymentService.<ConfirmPaymentAsync>d__11.MoveNext() in
```
Might be related to #712 but different trigger here.

**Comment by jennablackwell** [comment]
this exception started happening for me as well

**Comment by Leks12lk** [comment]
Hi,
I see this issue as well. 
It happens when attempt to use as query parameters those parameters that are not specified as a URL substitution.
So **this code does not work**
`[Get("api/someurl")]
Task<MyClass> Call([Header(""Authorization"")] string token, string param);`

**This code works:**
`[Get("api/someurl?param={param}")]
Task<MyClass> Call([Header(""Authorization"")] string token, string param);`

Please note that this issue is not reproduced for v4.7.9 but for 4.7.51 it's reproduced.

**Comment by SWarnberg** [comment]
It's a breaking change.

**Comment by Kopkins** [comment]
> Hi,
> I see this issue as well.
> It happens when attempt to use as query parameters those parameters that are not specified as a URL substitution.
> So **this code does not work**
> `[Get("api/someurl")] Task<MyClass> Call([Header(""Authorization"")] string token, string param);`
> 
> **This code works:**
> `[Get("api/someurl?param={param}")] Task<MyClass> Call([Header(""Authorization"")] string token, string param);`
> 
> Please note that this issue is not reproduced for v4.7.9 but for 4.7.51 it's reproduced.

I do see this happening in v4.7.9. The following code snippet reproduces the problem for me.
        `[Get("/documentcontent")]
        Task<DocumentContent> GetDocumentContent([Query] int documentId);`

**Comment by girtsl** [comment]
Any progress on this? Updating a *patch* version from 4.7.9 to 4.7.51 broke things for us, too. If this is by design, this is a breaking change. If it's a bug, it should likely be given a higher priority.

**Comment by onovotny** [comment]
@benjaminhowarth1 any luck with the fix to this issue?

@SWarnberg this is not intentional, it's a bug.

**Comment by girtsl** [comment]
@onovotny, thanks, I'm full aware of it. In case my previous comment came across as an arrogant one, it was not. It was merely meant as an indicator that the issue is not limited to a few obscure scenarios.
Seeing how this breaking change was discovered on August 30 and how easy it is to trip on it, I thought it would've attracted more attention from the maintainers who are in a much better position to find the cause and come up with a fix.

**Comment by onovotny** [comment]
Fixed by #772

**Comment by pbarrios** [comment]
This problem is still occurring in v5.0.15
